### ISSUE TITLE
Always compiling with debug

### ISSUE THREAD

**Issue by mhier** [ChimeraTK/open62541-interface#29]
It seems that the interface is compiling the stack always with Debug flags. This will create performance issues and also seems to cause issues with out TSAN and ASAM builds on Jenkins.

**Comment by zenker** [comment]
Thanks for reporting.
I agree we should change this to `Release`. I will change this together with another performance relevant bug fix that is in the pipeline and will be applied soon.

**Comment by killenb** [comment]
It should not be fixed. In Debug-build is should stay debug to get access to the Debug symbols. Ideally, we would also pass on tsan/asan flags. I don't know if this is possbile. Also, RelWithDebInfo can be quite useful. We use that for all of our debian packages and provide the debug symbols in a separate package. Again, I don't know if this is possible without changing the external package.

**Comment by zenker** [comment]
I can have a look how to fix that - if it is not possible to pass it via the external package mechanism I can add it to the @CMakeLists.txt@ directly via the patch we are already using...

**Comment by mhier** [comment]
We should also pass on the used compiler (our asan/tsan builds are using clang while the normal builds are using gcc).

**Comment by zenker** [comment]
How do you do that?
Usually I set some environment variables to let cmake pick clang at build time - do you want to fix the compiler for the asan/tsan build target?

Another thing that came to my mind is why would we want to ran tsan/asan on the open62541 stack - basically it is just an external dependency and the project itself is responsible to track memory issue and so on. Moreover there is an option to turn on asan flags so I expect that the open62541 guys are running tests with asan routinely.

Should I still provide those build target?

**Comment by mhier** [comment]
> How do you do that? Usually I set some environment variables to let cmake pick clang at build time - do you want to fix the compiler for the asan/tsan build target?

No, but the environment variables need to affect also the build of the stack itself. Maybe they already do?


> Another thing that came to my mind is why would we want to ran tsan/asan on the open62541 stack - basically it is just an external dependency and the project itself is responsible to track memory issue and so on. Moreover there is an option to turn on asan flags so I expect that the open62541 guys are running tests with asan routinely.

In theory you should compile all your dependencies with tsan/asan if you do such analysis, even including BOOST (which we cannot do easily and suffer from it already).
